tests/src/JIT/Performance/CodeQuality/BenchmarksGame/k-nucleotide/k-nucleotide.cs

```diff
@@ -0,0 +1,284 @@
+﻿/* The Computer Language Benchmarks Game
+   http://benchmarksgame.alioth.debian.org/
+ *
+ * submitted by Josh Goldfoot
+ * 
+ */
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+[assembly: OptimizeForBenchmarks]
+[assembly: MeasureInstructionsRetired]
+[assembly: MeasureGCCounts]
+
+namespace BenchmarksGame
+{
+
+public class knucleotide
+{
+#if DEBUG
+    const int Iterations = 1;
+    const string InputFile = "knucleotide-input.txt"; 
+#else
+    const int Iterations = 10;
+    const string InputFile = "knucleotide-input-big.txt"; 
+#endif
+
+
+    static string FindInput(string s)
+   {
+       string CoreRoot = System.Environment.GetEnvironmentVariable("CORE_ROOT");
+
+       if (CoreRoot == null)
+       {
+           Console.WriteLine("This benchmark requries CORE_ROOT to be set");
+           return null;
+       }
+
+       string inputFile = s ?? InputFile;
+
+       // Normal testing -- input file will end up next to the assembly
+       // and CoreRoot points at the test overlay dir
+       string[] pathPartsNormal = new string[] {
+           CoreRoot, "..", "..", "JIT", "Performance",
+           "CodeQuality", "BenchmarksGame", "k-nucleotide", inputFile
```

Comment:
@guhuro see https://github.com/dotnet/coreclr/pull/9281 - we need to change the way we're finding the input files. We can't rely on them always being in the same place relative to CORE_ROOT, since in Helix, the directory structure assumed here isn't maintained - so tests like this one are failing there (because of that assumption). I've tried looking relative to the executing assembly, but that doesn't work, since the executing assembly is CoreRun.exe. I've also tried finding the directory of the assembly defined in the class itself, but that didn't work either, because for whatever reason we don't have access to the ".Location" member of the "Assembly" class (not sure why that is...). If you have any bright ideas, I'm all ears - otherwise I'm still digging to find a solution.